raft-tech/TANF-app
# Feat/132 parser file to model, no validators

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #132 

## How to Test

```
export TDRS_HOME=$PWD
source commands.sh
tdrs-start-av
```

1. Open http://localhost:3000/ and sign in.
1. If not already done, setup user account and access data-files page.
1. Upload and submit sample data files for a TANF-only stt. 
2. Check logs that errors did (not) occur. 
3. Access admin panel and inspect if any t1 search_index objects were created.

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:

**Acceptance Criteria:** 
- [ ] List of lessons learned from tackling corner-cases to be applied to other parsing tickets

Old AC within other tickets:
- [ ] TDP gives the user all data errors for a submission in a way they can easily save or export so they can take the issues back to their case management software and fix the issues then recreate an export for resubmission to TDP. #1122 
- [ ] Data Errors must be issued in human readable plain language so that an English speaker can easily understand what is needed. #1122

**Tasks:**
- [x] Using test-driven development, write unit tests for badly formatted files
- [x] Implement parsing for base-case small, error-free file
- [x] Implement parsing to log error messages for use-cases for badly formatted files
- [x] Parsing classes/functions should return a data structure(s) with relevant fields containing data from the input file


### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
